KsApi/models/graphql/adapters/Project+FetchProjectRewardsByIdQueryData.swift

```diff
@@ -13,9 +13,56 @@ extension Project {
 
   static func projectRewards(from data: GraphAPI.FetchProjectRewardsByIdQuery.Data) -> [Reward] {
     let projectRewards = data.project?.rewards?.nodes?
-      .compactMap { $0?.fragments.rewardFragment }
-      .compactMap { Reward.reward(from: $0) } ?? []
+      .compactMap { node -> (GraphAPI.RewardFragment, [ShippingRule]?)? in
+        guard let rewardFragment = node?.fragments.rewardFragment else { return nil }
 
-    return projectRewards
+        // These shipping rules are constructed from simplified versions of the shipping rules.
+        // Names are not localized and should not be shown to users, but they contain the data we
+        // need to calculate shipping, just in a different initial format.
+        let expandedShippingRules = node?.simpleShippingRulesExpanded?
+          .compactMap { node -> ShippingRule? in
+            guard let node,
+                  let idString = node.locationId, let locationId = decompose(id: idString)
+            else {
+              return nil
+            }
+
+            let name = node.locationName ?? ""
+            let location = Location.init(
+              country: node.country,
+              displayableName: name,
+              id: locationId,
+              localizedName: name,
+              name: name
+            )
+
+            let cost = node.cost.flatMap(Double.init) ?? 0.0
+            let estimatedMin = Money(
+              amount: node.estimatedMin.flatMap(Double.init) ?? 0.0,
+              currency: Money.CurrencyCode(rawValue: node.currency ?? "")
+            )
+            let estimatedMax = Money.init(
+              amount: node.estimatedMax.flatMap(Double.init) ?? 0.0,
+              currency: Money.CurrencyCode(rawValue: node.currency ?? "")
+            )
+
+            return ShippingRule(
+              cost: cost,
+              id: nil,
```

Comment:
Should this have a nil id?

Reply:
Yeah, I think that's fine. The simpleShippingRule graphql type doesn't include an id (see https://github.com/kickstarter/kickstarter/blob/main/app/graph/query_types/simple_shipping_rule_type.rb#L4) and the `id` field of the shipping rule is nullable. We don't use the `id` field for anything useful - equality for shippingRules is based on `location`, for example. (We do accidentally use it for tests instead of `shippingRule.location.id` in a few places, though...)